promptulate/tools/sleep/tool.py

```diff
@@ -1,10 +1,10 @@
 import re
 from time import sleep
 
-from promptulate.tools import BaseTool
+from promptulate import Tool
 
 
-class SleepTool(BaseTool):
+class SleepTool(Tool):
```

Comment:
Use function declaration type Tool to define tool is a better choice if you want to optimize it.

Reply:
I think use oop way is better in the framework